### PR TITLE
Switch to using BERT features

### PR DESCRIPTION
**Pull request by brahmaneya**
Use average of bert embeddings in final layer to get features, and sklearn LR on top of the features to get an accuracy of 0.86 (as compared to ~0.72 with bag of ngrams).

**Test Plan**
tox -e crowd / style pass

### CODE REVIEW

File: crowdsourcing/crowdsourcing_tutorial.py [hunk 1]
```diff
@@ -260,39 +260,35 @@ def polarity_negative_2(x):
...
-_ = model.fit(X_train, Y_train_prob, epochs=10, verbose=0)
+sklearn_model = LogisticRegression()
+sklearn_model.fit(train_vectors, probs_to_preds(Y_train_prob))
 
-# %%
-probs = model.predict(X_test)
-preds = probs_to_preds(probs)
-acc = metric_score(Y_test, preds=preds, metric="accuracy")
-print(f"Test Accuracy when trained with soft training labels: {acc:.3f}")
+sklearn_model.score(test_vectors, Y_test)
```
Comment (henryre) [inline crowdsourcing/crowdsourcing_tutorial.py]: Nit: add a printout or markdown cell about test score

File: crowdsourcing/requirements.txt [hunk 2]
```diff
@@ -1,3 +1,4 @@
 textblob==0.15.3
 matplotlib==3.1.1    
 tensorflow==1.14.0
+pytorch-transformers==1.0.0
```
Comment (henryre) [inline crowdsourcing/requirements.txt]: Nit: can we alphabetize this?

File: crowdsourcing/crowdsourcing_tutorial.py [hunk 3]
```diff
@@ -260,39 +260,35 @@ def polarity_negative_2(x):
...
+    return model(input_ids)[0].mean(1)[0].detach().numpy()
+
+
+train_vectors = np.array(list(df_train.tweet_text.apply(encode_text).values))
+test_vectors = np.array(list(df_test.tweet_text.apply(encode_text).values))
 
 # %% [markdown]
 # ### Model on soft labels
-# Now, we train a simple MLP model on the bag-of-words features, using labels
+# Now, we train a simple Logistic Regression model on the BERT features, using labels
```
Comment (henryre) [inline crowdsourcing/crowdsourcing_tutorial.py]: nit: lower case log reg

File: crowdsourcing/crowdsourcing_tutorial.py [hunk 4]
```diff
@@ -260,39 +260,35 @@ def polarity_negative_2(x):
...
-
-# %% [markdown]
-# ### Featurization
+# ### Getting features from BERT
+# Since we have very limited training data, we cannot train a complex model like an LSTM with a lot of parameters. Instead, we use a pre-trained model, [BERT](https://github.com/google-research/bert), to generate embeddings for each our tweets, and treat the embedding values as features.
 
 # %%
-from sklearn.feature_extraction.text import CountVectorizer
+import torch
+from pytorch_transformers import BertModel, BertTokenizer
```
Comment (henryre) [inline crowdsourcing/crowdsourcing_tutorial.py]: nit:

import numpy as np
import torch
from pyto...

File: crowdsourcing/crowdsourcing_tutorial.py [hunk 5]
```diff
@@ -260,39 +260,35 @@ def polarity_negative_2(x):
...
 # %%
-import tensorflow as tf
+from sklearn.linear_model import LogisticRegression
 
-model = tf.keras.Sequential()
-model.add(tf.keras.layers.Dense(10, activation=tf.nn.relu))
-model.add(tf.keras.layers.Dense(2, activation=tf.nn.softmax))
-model.compile("Adam", "categorical_crossentropy")
-_ = model.fit(X_train, Y_train_prob, epochs=10, verbose=0)
+sklearn_model = LogisticRegression()
```
Comment (henryre) [inline crowdsourcing/crowdsourcing_tutorial.py]: nit: call `clf` for scikit convention

File: crowdsourcing/crowdsourcing_tutorial.py [hunk 6]
```diff
@@ -260,39 +260,35 @@ def polarity_negative_2(x):
...
 
 # %% [markdown]
 # ### Model on soft labels
-# Now, we train a simple MLP model on the bag-of-words features, using labels
+# Now, we train a simple Logistic Regression model on the BERT features, using labels
 # obtained from our label model.
 
 # %%
-import tensorflow as tf
+from sklearn.linear_model import LogisticRegression
```
Comment (henryre) [inline crowdsourcing/crowdsourcing_tutorial.py]: could we do any better with a little regularization? just an idea, honestly idk

**Comment by vincentschen** [comment]
High-level question— why train over BERT features in sklearn rather than using the BERT model or adding a linear layer to create a normal PyTorch model?

File: crowdsourcing/crowdsourcing_tutorial.py (hunk 6 above)
Comment (brahmaneya) [reply crowdsourcing/crowdsourcing_tutorial.py]: Tried l1 and l2 regularization. l1 does worse, while l2 does the same.

**Comment by brahmaneya** [comment]
> High-level question— why train over BERT features in sklearn rather than using the BERT model or adding a linear layer to create a normal PyTorch model?

The latter two options are not easily enabled by the pytorch-transformers repo (didn't check too thoroughly). But also, we have very only 187 examples, so training all ~100M parameters may not be that useful.
